### PR TITLE
Made the game kinda fun now

### PR DESCRIPTION
**Pull request by NegativeScript**
Only problem I see is no game progression. It doesn't get harder as time goes. Whatever that means.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -28,9 +28,42 @@ Game = {
...
+
+        path.splice(0, 1);
+        path.splice(path.length, 1);
+
         for (var x = 0; x < map.widthInTiles; x++) {
             for (var y = 0; y < map.heightInTiles; y++) {
-                map.getTile(x, y).setWallTile();
+                if (!isInArray(path, [x, y])) {
+                    var tile = map.getTile(x, y);
+                    var antiVirusno = 0;
```
Comment (ashes999) [inline src/main.js]: Should be `antiVirusNumber`

File: src/main.js [hunk 2]
```diff
@@ -28,9 +28,42 @@ Game = {
...
+                    switch (choice) {
+                        case 'StrongDangerTile':
+                            tile.setStrongDangerTile();
+                            break;
+                        case 'WeakDangerTile':
+                            tile.setWeakDangerTile();
+                            break;
+                        case 'AntiVirus':
+                            antiVirusno++;
+                            if (antiVirusno => 2) {
```
Comment (ashes999) [inline src/main.js]: Magic numbers like `2` should be in constants (or in config), why 2?

File: src/main.js [hunk 3]
```diff
@@ -28,9 +28,42 @@ Game = {
...
+            return false;
+        }
+
+        path.splice(0, 1);
+        path.splice(path.length, 1);
+
         for (var x = 0; x < map.widthInTiles; x++) {
             for (var y = 0; y < map.heightInTiles; y++) {
-                map.getTile(x, y).setWallTile();
+                if (!isInArray(path, [x, y])) {
```
Comment (ashes999) [inline src/main.js]: To add "difficulty," you can add a variable in config called `probabilityOfSpawningTile` (eg. 0.25 and `increaseSpawnProbabilityPerLevel` (eg. 0.05), up to `maxProbabilityOfSpwawning (eg. 0.75).

File: src/main.js (hunk 2 above)
Comment (NegativeScript) [reply src/main.js]: Has to deal with probability. I found that, without said variable, it spawns too many antiviruses.

File: config.json [hunk 4]
```diff
@@ -1,29 +1,44 @@
...
   "dangerDamage": 5,
   "antiVirusDamage": 5,
 
-  "allowAntiVirusEntities": false,
+  "allowAntiVirusEntities": true,
   "antiVirusMovementCost": 1,
-  "antiVirusesPerLevel": 1,
+
+  "mapSeed": "2828407160", // seed for generating the map
+  "pathNodes": 3,
```
Comment (ashes999) [inline config.json]: I would call this "pathSegments" but you can fix that later.

File: src/entities/antiVirus.js [hunk 5]
```diff
@@ -14,6 +14,10 @@ Crafty.c('AntiVirus', {
         }
         var tile = map.getTile(path[0], path[1]); // get tile from map
 
+        if (tile.entity == 'AntiVirus') {
+            return null;
```
Comment (ashes999) [inline src/entities/antiVirus.js]: Is this for antivirus-touching-antivirus? Maybe we can think of something interesting to do here.

File: src/entities/map.js [hunk 6]
```diff
@@ -28,13 +28,16 @@ map = {
 
     getReachableTile: function () {
         var tile = map.getRandomTile();
-        for (i = 0; i < this.heightInTiles * this.widthInTiles; i++) {
-            if (map.getPathToPlayer(tile).length != 0) {
-                return tile;
-            }
-            tile = map.getRandomTile()
+        var attempts = 0;
+        while (map.getPathToPlayer(tile).length == 0 && attempts < this.widthInTiles * this.heightInTiles) {
```
Comment (ashes999) [inline src/entities/map.js]: I would add `var maxAttempts = 100` instead so it's more understandable.

File: src/main.js [hunk 7]
```diff
@@ -28,30 +28,72 @@ Game = {
...
+        }
+
         for (var x = 0; x < map.widthInTiles; x++) {
             for (var y = 0; y < map.heightInTiles; y++) {
-                map.getTile(x, y).setWallTile();
+                if (!isInArray(path, [x, y]) && generator.random() < randomTileChance) {
+                    var tile = map.getTile(x, y);
+                    if (tile.contents == '' && tile.entity == '') {
+                        var antiVirusNumber = 0;
+                        var choice = generator.choice(['StrongDangerTile', 'WeakDangerTile', 'WallTile', 'WallTile', 'WallTile', 'WallTile', 'AntiVirus'])
```
Comment (ashes999) [inline src/main.js]: In the future, this block (from `var choice` to the end of the cases) would make a nice, small function to refactor out.

File: src/entities/antiVirus.js (hunk 5 above)
Comment (NegativeScript) [reply src/entities/antiVirus.js]: Yep. Previously, they'll just stack on the same tile, causing a lot of bugs. If you have a better suggestion, by all means.

Comment (ashes999) [reply src/entities/antiVirus.js]: We can brainstorm something.  Off the top of my head, two things come to mind:

- Fusion. Make them a "stronger" antivirus (?!) -- this makes sense if antiviruses tend to die after X number of moves (something we haven't implemented yet).
- Kill one or both. They collided. 

Feel free to brainstorm more, and add to the MVP list / config toggles / etc.

**Pull request merged**